sklearn/metrics/scorer.py

```diff
@@ -279,6 +289,9 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
         For example ``average_precision`` or the area under the roc curve
         can not be computed using discrete predictions alone.
 
+    name : str, default=None
+        The name for the scorer. If None, the score_func.__name__ is used.
```

Comment:
name should be a default arg, because `make_scorer` is public API and requires backwards compatibility. Were we redesigning this, it could be a required arg. You might as well make the default a string, rather than None.